### ISSUE TITLE
Upper Case Not Working

### ISSUE THREAD

**Issue by ebhills** [wrangleworks/WranglesPY#125]
when I ran upper in a recipe, I got NaNs all the way down.  Let me know if you can't repoduce this.

**Comment by ChrisWRWX** [comment]
Do you have some more context for this? What version of Wrangles are you using?

This recipe:
```yml
read:
  - test:
      rows: 5
      values:
        header: make upper case
      
wrangles:
  - convert.case:
      input: header
      case: upper

  - log: {}
```

Gives the expected output:
```
            header
0  MAKE UPPER CASE
1  MAKE UPPER CASE
2  MAKE UPPER CASE
3  MAKE UPPER CASE
4  MAKE UPPER CASE
```

**Comment by ebhills** [comment]
I was on old version but still didn't work aftger I upgraded.  then I found the issue. I was running it on a list, which is why it gave me NaNs.  So now I'd like to change the focus of this issue to the first of the 2 below:

1. Smarter Type Handling:  is there a way for all Wrangles that might get an unexpected data type to try to finesse it?  This and yesterday's example (mixed type of list and str) show how subtle differences can create problems.  It's probably just **Lists vs Strings** that we need to account for, so it doesn't seem it would be that hard to have a standard way of handling that automatically (vs having to add an additional Wrgls).
2. Errors: is there a way to a. capture errors and b. at least message back which Wrgl the error occurred on?  I realize this is proably a big discussion so we can pick that topic up elsewhere.
